File: Java/프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.HashMap;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        
        HashMap<String, Integer> targetMap = new HashMap<>();
        for (int i = 0; i < want.length; i++) {
            targetMap.put(want[i], number[i]);
        }
        
        int answer = 0;
        for (int i = 0; i < discount.length - 9; i++) {
            HashMap<String, Integer> items = new HashMap<>();
            
            for (int j = i; j < i + 10; j++) {
                String item = discount[j];
               if (targetMap.containsKey(item)){
                   items.put(item, items.getOrDefault(item, 0) + 1);
               }
            }
            
            if  (items.equals(targetMap)) {
                answer ++;
            }
        }
        
        return answer;
    }
}

File: Swift/프로그래머스/2/131127. 할인 행사/할인 행사.swift
import Foundation

func solution(_ want:[String], _ number:[Int], _ discount:[String]) -> Int {
    var target: [String: Int] = [:]
    
    for i in 0 ..< want.count {
        target[want[i]] = number[i]
    }
    
    var answer = 0
    for i in 0 ..< discount.count-9 {
        var itemsFor10d: [String: Int] = [:]
        for j in i ..< i+10 {
            let item = discount[j]
            if let _ = target[item] {
                itemsFor10d[item] = (itemsFor10d[item] ?? 0) + 1
            }
        }
        
        if itemsFor10d == target {
            answer += 1
        }
    }
    return answer
}